src/assessments/seo/titleKeywordAssessment.js

```diff
@@ -1,46 +1,113 @@
  */
-var titleHasKeywordAssessment = function( paper, researcher, i18n ) {
-	var keywordMatches = researcher.getResearch( "findKeywordInPageTitle" );
-	var score, text;
-
-	if ( keywordMatches.matches === 0 ) {
-		score = 2;
-		text = i18n.sprintf( i18n.dgettext( "js-text-analysis", "The focus keyword '%1$s' does " +
-			"not appear in the SEO title." ), escape( paper.getKeyword() ) );
+class TitleKeywordAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			parameters: {
+				recommendedMinimum: 1,
+				position: 0,
+			},
+			good: {
+				score: 9,
+				resultText: "The SEO title contains the focus keyword, at the beginning which is considered to " +
+				"improve rankings.",
+				requiresKeyword: false,
+			},
+			okay: {
+				score: 6,
+				resultText: "The SEO title contains the focus keyword, but it does not appear at the beginning; " +
+				"try and move it to the beginning.",
+				requiresKeyword: false,
+			},
+			bad: {
+				score: 2,
+				resultText: "The focus keyword '%1$s' does not appear in the SEO title.",
+				requiresKeyword: true,
+			},
+		};
+
+		this.identifier = "titleKeyword";
+		this._config = merge( defaultConfig, config );
 	}
 
-	if ( keywordMatches.matches > 0 && keywordMatches.position === 0 ) {
-		score = 9;
-		text = i18n.dgettext( "js-text-analysis", "The SEO title contains the focus keyword, at the beginning which is considered " +
-			"to improve rankings." );
+	/**
+	 * Executes the pagetitle keyword assessment and returns an assessment result.
+	 *
+	 * @param {Paper} paper The Paper object to assess.
+	 * @param {Researcher} researcher The Researcher object containing all available researches.
+	 * @param {Object} i18n The locale object.
+	 *
+	 * @returns {AssessmentResult} The result of the assessment with text and score
+	 */
+	getResult( paper, researcher, i18n ) {
+		this._keywordMatches = researcher.getResearch( "findKeywordInPageTitle" );
+
+		let assessmentResult = new AssessmentResult();
+
+		if ( paper.hasKeyword() && paper.hasTitle() ) {
+			const calculatedScore = this.calculateScore();
+			assessmentResult.setScore( calculatedScore.score );
+			assessmentResult.setText( this.translateScore( calculatedScore.resultText, calculatedScore.requiresKeyword,
+				escape( paper.getKeyword() ), i18n ) );
+		}
+
+		return assessmentResult;
 	}
 
-	if ( keywordMatches.matches > 0 && keywordMatches.position > 0 ) {
-		score = 6;
-		text = i18n.dgettext( "js-text-analysis", "The SEO title contains the focus keyword, but it does not appear at the beginning;" +
-			" try and move it to the beginning." );
+	/**
+	 * Calculates the result based on the keyphraseLength research.
+	 *
+	 * @returns {Object} object with score and text.
+	 */
+	calculateScore() {
```

Comment:
calculateResult? (in line with the previous assessments that have been altered like this)

Reply:
Agreed.